File: week1/day5/tandem.js
function tandemBicycle(red, blue, fastest) {
  if (fastest) {
    let all = [...red, ...blue];
    all = all.sort(function (a, b) {
      return a - b;
    });
    let length = red.length;
    let answer = all.slice(length).reduce((a, b) => a + b, 0);
    return answer;
  } else {
    red = red.sort(function (a, b) {
      return a - b;
    });
    blue = blue.sort(function (a, b) {
      return a - b;
    });
    let slowest = [];
    for (i = 0; i < red.length; i++) {
      speed = Math.max(red[i], blue[i]);
      slowest = [...slowest, speed];
      answer = slowest.reduce((a, b) => a + b, 0);
    }
    return answer;
  }
}

console.log(tandemBicycle([5, 5, 3, 9, 2], [3, 6, 7, 2, 1], false));

File: week1/day5/tandem.py
def tandemBicycle(red, blue, fastest):  
  answer = 0
  if fastest:
      all = red + blue
      all.sort(reverse=True)
      highestSpeeds = all[0:len(red)]
      for speed in highestSpeeds:
          answer = answer + speed
  else:
      red.sort()
      blue.sort()
      counter = len(red) -1
      while counter >= 0:
          speed = max(red[counter], blue[counter])
          answer = answer + speed
          counter -=1
  return answer

print(tandemBicycle([5,5,3,9,2], [3,6,7,2,1], True))
print(tandemBicycle([5,5,3,9,2], [3,6,7,2,1], False))


